packages/core/src/shared/telemetry/activation.ts

```diff
@@ -49,12 +50,23 @@ export async function activate(
     globals.telemetry = await DefaultTelemetryService.create(extensionContext, awsContext, getComputeRegion())
 
     try {
-        globals.telemetry.telemetryEnabled = config.isEnabled()
+        await globals.telemetry.setTelemetryEnabled(config.isEnabled())
 
         extensionContext.subscriptions.push(
-            (isAmazonQ() ? config.amazonQConfig : config.toolkitConfig).onDidChange(event => {
+            (isAmazonQ() ? config.amazonQConfig : config.toolkitConfig).onDidChange(async event => {
                 if (event.key === 'telemetry') {
-                    globals.telemetry.telemetryEnabled = config.isEnabled()
+                    const val = config.isEnabled()
+                    if (!val) {
+                        // Record 'disabled' right before its turned off, so we can send this + the batch we have already.
+                        telemetry.ui_click.emit({ elementId: 'aws_disableTelemetry' })
```

Comment:
`aws_modifySetting` is an alternative to `ui_click` : https://github.com/aws/aws-toolkit-common/blob/9fdefa93c1741bf2cc3046a01f70fd7cc3eed540/telemetry/definitions/commonDefinitions.json#L2343

Reply:
Ohh I missed it. Thanks Justin! let's change it to 
`aws_modifySetting.emit({ settingId: 'telemetryEnabled', settingState: true/false })`

when settingState is `false` it means telemetry is disabled 
when settingState is `true` it means telemetry is enabled 
<img width="527" alt="image" src="https://github.com/aws/aws-toolkit-vscode/assets/50885351/b77e83e1-b735-4267-af61-b9ef4cb38617">